chapter_1/kmeans_hardness_d2_k2.tex

```diff
@@ -0,0 +1,150 @@
+
+Definition 5 allows us to see where the number of clusters $k=l(3n+2)+(l-1)3n$ comes from. A nice clustering corresponds naturally to this $k$, since there are $l-1$ $M$ rows, each with $3n$ points in singleton clusters, and $l$ $R$ rows, with $3n+2$ clusters each, corresponding to $3n+1$ clusters with two consecutive points and 1 cluster with one point (whether $s_i$ or $f_i$ is the singleton cluster depends on whether $R_i$ is in an $A$ or $B$ clustering, respectively). Note an $A$ clustering saves exactly $\alpha$ cost compared to a $B$ clustering.
+
+Let's discuss \textbf{Lemma 7}. The purpose of this lemma is to establish the fact that for $w$ large enough, any optimal clustering of $H_{l,n}$ must be ``nice.'' In fact, Lemma 6 allows us to make the stronger statement that the optimal clustering will be nice \textit{and} all $R_i$'s will be grouped in $A$ clusterings, for a total cost of $L_1-l\alpha$.
+
+(However, soon we will be adding more points to create $G_{l,n}\cup X$, and for this augmented instance, we will argue that iff the corresponding X3C instance is satisfiable, the optimal clustering will have exactly $n$ $R_i$ in the $A$ clustering and the remaining $R_i$ in the $B$ clustering.)
+
+The proof of Lemma 7 is simply via casework and calculating the costs for each case.
+
+Now let's construct $G_{l,n}\cup X$, the $k$-means instance corresponding to the reduction. Let's read up until \textbf{Definition 8 and Lemma 9}. Note that the points we are adding to $H_{l,n}$ correspond to $\cC$ from X3C. The construction seem a bit strange, but just note for now that (1) we have added $6n(l-1)$ points and (2) if $j\in S_i$, then the corresponding $x'_{i,j}$ and $y'_{i-1,j}$ will only have a ``good'' cluster in $R_i$ if $R_i$ has a $B$ clustering, whereas $j\not\in S_i$ does not impose this restriction on the corresponding $x_{i,j}$ and $y_{i-1,j}$, i.e. $R_i$ can have either an $A$ or $B$ clustering.
+
+Definition 8 and Lemma 9 justify the choice of $L_2=6n(l-1)h^2\frac{2w}{2w+1}$; looking at the figure, we can see that the ``good" clusters mentioned in Lemma 9 are indeed the closest clusters, and minimize the cost of each of the $6n(l-1)$ new points to be $h^2\frac{2w}{2w+1}$, if they are added to one of these existing clusters.
+
+\textbf{Lemma 10} goes further and states that any optimal $k$-clustering of $G_{l,n}\cup X$ must retain the ``nice'' clustering on the $H_{l,n}$ instance, and further, each $x\in X$ must be assigned to a different ``good'' cluster. The proofs of Lemmas 9 and 10 simply add up the costs according to the construction.
+
+Now we're ready for the reduction. I think this part was the trickiest to understand, so I'll try to provide some intuition.
+
+\textbf{Lemma 11.} \textit{The set $G_{l,n}\cup X$ has a $k$-clustering of cost less or equal to $L$ if and only if there is an exact cover $\mathcal{F}\subseteq \cC$ for the X3C instance.}
+
+$(\implies)$ Suppose we have an optimal $k$-clustering with cost $\le L$. The
```

Comment:
Is this the proof of lemma 11? If so put it in \begin{proof} environment or just state you will now prove lemma 11. It's a little unclear as is.